src/manifest/representation.ts

```diff
@@ -159,7 +173,8 @@ class Representation {
         }
         const initData = concat(...initDataArr.map(({ data }) => data));
         acc.push({ type: initDataType,
-                   data: initData });
+                   data: initData,
+                   keyIds: this.contentProtections?.keyIds.map(val => val.keyId) });
```

Comment:
This is also an ugly part.
I think that content protection data stored with a `Representation` should directly be stored in the exploitable format stored here. I'm working on it.

Reply:
After re-reading. I think the code is actually OK.
I still moved that line higher up.